internal/x11/win/frame.go

```diff
@@ -484,16 +485,31 @@ func (f *frame) maximizeApply() {
 }
 
 func (f *frame) mouseDrag(x, y int16) {
-	if f.client.Maximized() || f.client.Fullscreened() {
+	if f.client.Fullscreened() {
 		return
 	}
 	moveDeltaX := x - f.mouseX
 	moveDeltaY := y - f.mouseY
-	f.mouseX = x
-	f.mouseY = y
 	if moveDeltaX == 0 && moveDeltaY == 0 {
 		return
 	}
+	if f.client.Maximized() {
+		if uint16(moveDeltaX) > x11.TitleHeight(x11.XWin(f.client)) ||
```

Comment:
Might it be clearer if we add a new variable that avoids this call twice? Might help show intent too.
```go
dragThreshold := x11.TitleHeight(x11.XWin(f.client))
```

Reply:
With the new changes it's only one threshold now, vertical.  Per @Jacalz horizontal drag shouldn't unmaximize.  That will change we do quarter snapping though I'm sure.

Reply:
There is now a magic "*3" so I think the comment still applies with regard to intent.
Looking at the code it is also not clear why the y position needs to be checked twice.

Reply:
Thanks for the explanation, if the first check is to ensure it is within the bar then you’re missing a check for y >= f.y.

I don’t think a const makes sense, as the title height is a variable.
However the same border theme lookup is being done twice in this line, so is it not more efficient to capture in a var?